lib/shard/src/proxy_segment/segment_entry.rs

```diff
@@ -36,7 +36,17 @@ impl SegmentEntry for ProxySegment {
             .get()
             .read()
             .point_version(point_id)
-            .or_else(|| self.wrapped_segment.get().read().point_version(point_id))
+            .or_else(|| {
+                // Ignore point from wrapped segment if already marked for deletion
+                // By `point_version` semantics we don't expect to get a version if the point
+                // is deleted. This also prevents `move_if_exists` from moving an old point
+                // into the write segment again.
+                if self.deleted_points.read().contains_key(&point_id) {
+                    return None;
```

Comment:
Don't we need to check that the deleted_version is > than the version found in the wrapped segment?

Reply:
BTW my local change is

```rust
let wrapped_version = self.wrapped_segment.get().read().point_version(point_id);
let deleted_version = self.deleted_points.read().get(&point_id).cloned();
wrapped_version.zip(deleted_version).and_then(|(wrapped_version, deleted_version)| {
    if deleted_version.local_version < wrapped_version {
        Some(wrapped_version)
    } else {
        None
    }
})
```